plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -177,7 +190,13 @@ internal fun Package.toSpdxPackage(
             SpdxPackageType.VCS_PACKAGE -> SpdxConstants.NOASSERTION
             else -> concludedLicense.nullOrBlankToSpdxNoassertionOrNone()
         },
-        licenseDeclared = declaredLicensesProcessed.toSpdxDeclaredLicense(),
+        licenseDeclared = if (packageLicenseExpressions.isEmpty()) {
+            SpdxConstants.NONE
+        } else {
+            packageLicenseExpressions.reduce(SpdxExpression::and)
+                .sorted()
+                .nullOrBlankToSpdxNoassertionOrNone()
+        },
```

Comment:
Should the license choices be applied somehow?

Reply:
Good point. I'll probably do that in a separate commit as that was already missing the the original code for `licenseInfoFromFiles`, which should get choices applied as well.

Reply:
Yes, I just wanted to point this out. I believe this is a rather complex open topic, because of current semantics of the choices. I'm fine if you consider this out of scope too)

FYI: I believe semantics of choice should be changed, so that we can apply them to each single detected license finding. Not to the joined union of all detections and so forth, but again that a bigger topic.

Reply:
> so that we can apply them to each single detected license finding.

As a side-note, this should already be possible. Just first filter with a specific `LicenseView` (and eventually also more filters), and *then* apply the choice. Like

    licenseInfoResolver.resolveLicenseInfo(id)
        .filter(LicenseView.ONLY_DETECTED)
        // Use maybe more filters, like by path.
        .applyChoices(...)
        .forEach { /* Do something with each resolved license. */ }